scripts/train/train.py

```diff
@@ -164,170 +164,266 @@ def build_dataloader(cfg: DictConfig, tokenizer: PreTrainedTokenizerBase,
             tokenizer,
             device_batch_size,
         )
-
     else:
         raise ValueError(f'Not sure how to build dataloader with config: {cfg}')
 
 
-def main(cfg: DictConfig):
-    # Check for incompatibilities between the model and data loaders
-    validate_config(cfg)
+def pop_config(cfg: DictConfig,
+               key: str,
+               must_exist: bool = True,
+               default_value: Any = None) -> Any:
+    """Pop a value from the main config file and return it.
+
+    If the key does not exist, return the default_value or raise a RuntimeError
+    depending on the must_exist flag.
+    """
+    value = cfg.pop(key, None)
+    if value:
+        return value
+    elif must_exist:
+        raise RuntimeError(
+            f'The {key} parameter is missing and must exist for execution. Please check your yaml.'
+        )
+    else:
+        warnings.warn(
+            f'No {key} parameters specified. {key} will not be used during execution.'
+        )
+        return default_value
+
 
+def main(cfg: DictConfig):
     # Filter deprecation warning from torch internal usage
     warnings.filterwarnings(
         action='ignore',
         category=UserWarning,
         message=
-        f'torch.distributed.*_base is a private function and will be deprecated.*'
+        'torch.distributed.*_base is a private function and will be deprecated.*'
     )
 
-    cfg.dist_timeout = cfg.get('dist_timeout', 600.0)
+    # Check for incompatibilities between the model and data loaders
+    validate_config(cfg)
+
+    # Resolve all interpolation variables as early as possible
+    om.resolve(cfg)
```

Comment:
did you test this line? I thought previously when I did this I had to do `om.to_container(cfg, resolve=True)`, but maybe that was because I needed an actual dict out. Should we just convert to an actual dict here?